File: benchmarks/Java_Clojure_Set_Primitives/sort_m2_data1000.java
import de.hhu.stups.btypes.BRelation;
import de.hhu.stups.btypes.BSet;
import de.hhu.stups.btypes.BTuple;
import de.hhu.stups.btypes.BInteger;
import de.hhu.stups.btypes.BBoolean;
import de.hhu.stups.btypes.BUtils;

public class sort_m2_data1000 {

    private final BRelation<BInteger, BInteger> f;
    private final BInteger n;




    private BInteger j;
    private BInteger k;
    private BInteger l;
    private BRelation<BInteger, BInteger> g;

    public sort_m2_data1000() {
        BRelation<BInteger, BInteger> _ic_set_0 = new BRelation<BInteger, BInteger>();
        for(BInteger _ic_i : BSet.interval(new BInteger(1),new BInteger(1000))) {
            _ic_set_0 = _ic_set_0.union(new BRelation<BInteger, BInteger>(new BTuple<>(_ic_i, new BInteger(1500).minus(_ic_i))));

        }
        f = _ic_set_0;
        n = new BInteger(1000);
        g = f;
        k = new BInteger(1);
        l = new BInteger(1);
        j = new BInteger(1);
    }

    public void progress() {
        if((k.unequal(n).and(j.equal(n))).booleanValue()) {
            BRelation<BInteger, BInteger> _ld_g = g;
            BInteger _ld_k = k;
            BInteger _ld_l = l;
            g = _ld_g.override(new BRelation<BInteger, BInteger>(new BTuple<>(_ld_k, _ld_g.functionCall(_ld_l))).override(new BRelation<BInteger, BInteger>(new BTuple<>(_ld_l, _ld_g.functionCall(_ld_k)))));
            k = _ld_k.plus(new BInteger(1));
            j = _ld_k.plus(new BInteger(1));
            l = _ld_k.plus(new BInteger(1));
        }
    }

    public void prog1() {
        if((k.unequal(n).and(j.unequal(n)).and(g.functionCall(l).lessEqual(g.functionCall(j.plus(new BInteger(1)))))).booleanValue()) {
            BInteger _ld_j = j;
            BInteger _ld_l = l;
            l = _ld_l;
            j = _ld_j.plus(new BInteger(1));
        }
    }

    public void prog2() {
        if((k.unequal(n).and(j.unequal(n)).and(g.functionCall(l).greater(g.functionCall(j.plus(new BInteger(1)))))).booleanValue()) {
            BInteger _ld_j = j;
            j = _ld_j.plus(new BInteger(1));
            l = _ld_j.plus(new BInteger(1));
        }
    }

    public void final_evt() {
        if((k.equal(n)).booleanValue()) {
        }
    }



}

File: benchmarks/execution/Python_Primitives/sort_m2_data1000.py
from btypes.BRelation import BRelation
from btypes.BSet import BSet
from btypes.BTuple import BTuple
from btypes.BInteger import BInteger
from btypes.BBoolean import BBoolean



class sort_m2_data1000:
    def __init__(self):
        self._ic_set_0 = BRelation()
        for _ic_i in BSet.interval(BInteger(1), BInteger(1000)):
            self._ic_set_0 = self._ic_set_0.union(BRelation(BTuple(_ic_i, BInteger(1500).minus(_ic_i))))
        self.f = self._ic_set_0
        self.n = BInteger(1000)
        self.g = self.f
        self.k = BInteger(1)
        self.l = BInteger(1)
        self.j = BInteger(1)

    def progress(self):
        if BBoolean(self.k.unequal(self.n).booleanValue() and self.j.equal(self.n).booleanValue()).booleanValue():
            _ld_g = self.g
            _ld_k = self.k
            _ld_l = self.l
            self.g = _ld_g.override(BRelation(BTuple(_ld_k, _ld_g.functionCall(_ld_l))).override(BRelation(BTuple(_ld_l, _ld_g.functionCall(_ld_k)))))
            self.k = _ld_k.plus(BInteger(1))
            self.j = _ld_k.plus(BInteger(1))
            self.l = _ld_k.plus(BInteger(1))

    def prog1(self):
        if (BBoolean(BBoolean(self.k.unequal(self.n).booleanValue() and
                              self.j.unequal(self.n).booleanValue()).booleanValue() and
                              self.g.functionCall(self.l).lessEqual(self.g.functionCall(self.j.plus(BInteger(1)))).booleanValue())).booleanValue():
            _ld_j = self.j
            _ld_l = self.l
            self.l = _ld_l
            self.j = _ld_j.plus(BInteger(1))

    def prog2(self):
        if BBoolean(BBoolean(self.k.unequal(self.n).booleanValue() and
                             self.j.unequal(self.n).booleanValue()).booleanValue() and
                             self.g.functionCall(self.l).greater(self.g.functionCall(self.j.plus(BInteger(1)))).booleanValue()).booleanValue():
            _ld_j = self.j
            self.j = _ld_j.plus(BInteger(1))
            self.l = _ld_j.plus(BInteger(1))

    def final_evt(self):
        if (self.k.equal(self.n)).booleanValue():
            pass
